Content.Client/Chat/TypingIndicatorSystem.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using Content.Client.Chat.UI;
+using Content.Shared.CCVar;
+using Content.Shared.Chat;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Console;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Timing;
+
+namespace Content.Client.Chat
+{
+    public class TypingIndicatorSystem : SharedTypingIndicatorSystem
+    {
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+        [Dependency] private readonly IEyeManager _eyeManager = default!;
+        [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+        private readonly Dictionary<EntityUid, TypingIndicatorGui> _guis = new();
+        private IEntity? _attachedEntity;
+
+        /// <summary>
+        /// The system needs to be enabled by default client side for handling remote players
+        /// but clients can opt in to use this feature.
+        /// </summary>
+		public bool EnabledLocally => _cfg.GetCVar(CCVars.ChatTypingIndicatorSystemEnabled);
+
+        /// <summary>
+        /// Used for throttling how often we allow the client
+        /// to send messages informing the server they are typing.
+        /// </summary>
+        private bool _canTransmit = false;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _canTransmit = true;
+            SubscribeNetworkEvent<RemoteClientTypingMessage>(HandleRemoteTyping);
+            SubscribeLocalEvent<PlayerAttachSysMessage>(HandlePlayerAttached);
+        }
+
+        public void HandleClientTyping()
+        {
+            if (!EnabledLocally || !_canTransmit) return;
+            RaiseNetworkEvent(new ClientTypingMessage(_playerManager?.LocalPlayer?.UserId, _attachedEntity?.Uid));
+            _canTransmit = false;
+            Timer.Spawn(1000 * (int) _cfg.GetCVar(CCVars.ChatTypingIndicatorPollRate), () =>
+            {
+                _canTransmit = true;
+            });
+        }
+
+        private void HandleRemoteTyping(RemoteClientTypingMessage ev)
+        {
+            var entity = EntityManager.GetEntity(ev.EnityId.GetValueOrDefault());
+            var comp = entity.EnsureComponent<TypingIndicatorComponent>();
+
+            //a remote client is typing. toggle their component so we can render the overlay locally.
+            comp.IsVisible = true;
+            Timer.Spawn(3000, () =>
+            {
+                comp.IsVisible = false;
+            });
+        }
```

Comment:
Just use a component and states instead of this

Reply:
isnt that what im doing? if not please link me an example.